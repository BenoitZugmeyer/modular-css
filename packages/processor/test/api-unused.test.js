"use strict";

const namer = require("@modular-css/test-utils/namer.js");
const relative = require("@modular-css/test-utils/relative.js");

const Processor = require("../processor.js");

describe("/processor.js", () => {
    describe(".unused()", () => {
        let processor;

        beforeEach(() => {
            processor = new Processor({
                namer,
            });
        });

        it("should return a map of unused exports", async () => {
            await processor.string(
                "./one.css",
                ".one1 { } .one2 { }",
            );
            
            await processor.string(
                "./two.css",
                ".two1 { } .two2 { }",
            );

            processor._used(processor._normalize("./one.css"), "one1");
            processor._used(processor._normalize("./two.css"), "two2");

            const unused = processor.unused();

            expect(unused).toBeInstanceOf(Map);

            // Absolute file paths hork things up, so need to transform this
            const expected = [ ...unused.entries() ].map(([ file, classes ]) => [
                relative(file)[0],
                classes,
            ]);
            
            expect(expected).toMatchSnapshot();
        });
    });
});
