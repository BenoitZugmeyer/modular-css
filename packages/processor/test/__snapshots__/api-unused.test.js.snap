// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js .unused() should return a map of unused exports 1`] = `
Array [
  Array [
    "packages/processor/test/specimens/global.css",
    Set {
      "global",
    },
  ],
  Array [
    "packages/processor/test/specimens/composes-global.css",
    Set {
      "one",
    },
  ],
  Array [
    "packages/processor/test/specimens/blue.css",
    Set {
      "wooga",
    },
  ],
  Array [
    "packages/processor/test/specimens/start.css",
    Set {
      "booga",
      "tooga",
    },
  ],
  Array [
    "packages/processor/test/specimens/local.css",
    Set {
      "looga",
    },
  ],
  Array [
    "packages/processor/test/specimens/folder/folder.css",
    Set {
      "folder",
    },
  ],
]
`;
